### PR TITLE
Add development readme

### PR DESCRIPTION
**Pull request by nmuesch**
Creates a DEVELOPMENT.md file that showcases a quick way to get started with developing on this project.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## Developing on this Jenkins Plugin
+
+This page will outline some example steps that can be taken to spin up a development enviornment for this repository. The requirements are:
+
+* Java 1.8
+* docker-compose
```
Comment (ofek) [inline DEVELOPMENT.md]:
```suggestion
* Docker & docker-compose
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    image: jenkins/jenkins:lts
+    ports:
+      - 8080:8080
+    volumes:
+      - $JENKINS_PLUGIN/target/:/var/jenkins_home/plugins
+```
+
+Set the JENKINS_PLUGIN environment variable to point to the folder of the clone or fork of this repository.
+
+From here you can run `docker-compose up -f `<path_to_docker_compose_file>`.
```
Comment (ofek) [inline DEVELOPMENT.md]:
```suggestion
From here you can run `docker-compose -f `<path_to_docker_compose_file> up -d`.
```

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+## Developing on this Jenkins Plugin
+
+This page will outline some example steps that can be taken to spin up a development enviornment for this repository. The requirements are:
```
Comment (dabcoder) [inline DEVELOPMENT.md]: `enviornment` ==> `environment`. Just a minor typo.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## Developing on this Jenkins Plugin
+
+This page will outline some example steps that can be taken to spin up a development environment for this repository. The requirements are:
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
This page outlines some example steps to spin up a development environment for the *jenkins-datadog-plugin* repository. The requirements are:
```

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+* Java 1.8
+* Docker & docker-compose
+* A clone/fork of this repository
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
* [A clone/fork of this repository](https://help.github.com/en/articles/fork-a-repo)
```

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+## Developing on this Jenkins Plugin
+
+This page will outline some example steps that can be taken to spin up a development environment for this repository. The requirements are:
+
+* Java 1.8
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
* [Java 1.8](https://www.java.com/en/download/)
```

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+* Docker & docker-compose
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
* [Docker](https://docs.docker.com/get-started/) & [docker-compose](https://docs.docker.com/compose/install/)
```

File: DEVELOPMENT.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+* A clone/fork of this repository
+
+
+To get started, save the following docker-compose file somewhere locally:
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
To get started, save the following `docker-compose.yaml` file in your working directory locally:
```

File: DEVELOPMENT.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+services:
+  jenkins:
+    image: jenkins/jenkins:lts
+    ports:
+      - 8080:8080
+    volumes:
+      - $JENKINS_PLUGIN/target/:/var/jenkins_home/plugins
+```
+
+Set the JENKINS_PLUGIN environment variable to point to the folder of the clone or fork of this repository.
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
Set the `JENKINS_PLUGIN` environment variable to point to the folder of the clone/fork of this repository.
```

File: DEVELOPMENT.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+From here you can run `docker-compose -f `<path_to_docker_compose_file> up -d`.
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
Then run `docker-compose -f <DOCKER_COMPOSE_FILE_PATH> up -d`.
```

File: DEVELOPMENT.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+This will spin up the Jenkins docker image and auto mount the target folder of this repository (the location where the binary is built)
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
This spins up the Jenkins docker image and auto mount the target folder of this repository (the location where the binary is built)
```

File: DEVELOPMENT.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+To see updates, after re building the provider with `mvn clean package` on your local machine, simply run `docker-compose down` and spin this up again.
```
Comment (l0k0ms) [inline DEVELOPMENT.md]:
```suggestion
To see updates, after re building the provider with `mvn clean package` on your local machine, run `docker-compose down` and spin this up again.
```

**Pull request merged**